src/Microsoft.OpenApi/Extensions/EnumExtensions.cs

```diff
@@ -21,6 +23,7 @@ public static class EnumExtensions
         /// <returns>
         /// The attribute of the specified type or null.
         /// </returns>
+        [RequiresUnreferencedCode("GetAttributeOfType is not trim-compatible. Recommended to use native AoT-friendly type-specific overloads of GetDisplayName instead.")]
```

Comment:
> Recommended to use native AoT-friendly type-specific overloads of GetDisplayName instead.

It seems like this recommendation doesn't apply when the caller is trying to get an attribute from the enum.

Reply:
It doesn't, but based on @baywet's feedback in https://github.com/microsoft/OpenAPI.NET/pull/1717#discussion_r1681331142 we can obsolete this for now.

IMO, this is something that should be handled by the OpenAPI-layer in ASP.NET Core or other implementors, especially because things can get interesting depending on if you have `JsonStringEnumConverter` enabled.